internal/registry/registry.go

```diff
@@ -39,14 +39,16 @@ type DockerConfigJSON struct {
 }
 
 type ConnectionDetails struct {
-	*DockerConfigJSON
-	Namespace string
+	RegistryCredentials []RegistryCredentials
```

Comment:
Many of the changes in this PR are cause by this change here. I wanted to change the ConnectionDetails like this, because the json representation of the credentials is only important at the last minute, when we store the details as a Secret (or when we do the opposite). It feels more natural to work with an array of RegistryCredentials everywhere else. E.g. in `getInternalRegistryDetails`.